builders/tools/sidecar.md

```diff
@@ -143,36 +200,98 @@ To obtain the EVM sender address, recipient address, and EVM hash, check the `ev
+    | `EVM Hash`             | extrinsics.{extrinsic number}.events.{event number}.data.2 |
+    | `EVM Execution Status` | extrinsics.{extrinsic number}.events.{event number}.data.3 |
+
+=== "EIP2930"
+    | EVM Field              | Block JSON Field                                           |
+    |------------------------|------------------------------------------------------------|
+    | `Chain ID`             | extrinsics.{extrinsic number}.args.transaction.eip2930.chainId|
+    | `Nonce`                | extrinsics.{extrinsic number}.args.transaction.eip2930.nonce |
+    | `GasPrice`             | extrinsics.{extrinsic number}.args.transaction.eip2930.gasPrice|
+    | `GasLimit`             | extrinsics.{extrinsic number}.args.transaction.eip2930.gasLimit|
+    | `Value`                | extrinsics.{extrinsic number}.args.transaction.eip2930.value| 
+    | `Access List`          | extrinsics.{extrinsic number}.args.transaction.eip2930.accessList|       
+    | `Signature`            | extrinsics.{extrinsic number}.args.transaction.eip2930.oddYParity/r/s|
+    | `Sender Address`       | extrinsics.{extrinsic number}.events.{event number}.data.0 |
+    | `Recipient Address`    | extrinsics.{extrinsic number}.events.{event number}.data.1 |
+    | `EVM Hash`             | extrinsics.{extrinsic number}.events.{event number}.data.2 |
+    | `EVM Execution Status` | extrinsics.{extrinsic number}.events.{event number}.data.3 |
+
 
 !!! note
-    EVM transaction nonce and signature fields are under `extrinsics.{extrinsic number}.args.transaction`, whereas the `nonce` and `signature` fields under `extrinsics.{extrinsic number}` are the Substrate transaction nonce and signature, which are set to `null` for EVM transactions.
+    EVM transaction nonce and signature fields are under `extrinsics.{extrinsic number}.args.transaction.{transactiontype}`, whereas the `nonce` and `signature` fields under `extrinsics.{extrinsic number}` are the Substrate transaction nonce and signature, which are set to `null` for EVM transactions.
 
 
-### Computing Gas Spent {: #computing-gas-used } 
+### Computing Gas Used {: #computing-gas-used } 
 
 To calculate the gas spent or used during EVM execution of the transaction, the following formula can be used: 
 
-```
-GasPrice * Tx Weight / {{ networks.moonbase.tx_weight_to_gas_ratio }}
-```
 
-where GasPrice can be read from the block according to the above table, and Tx Weight can be retrieved under the event of the relevant extrinsic where the `method` field is set to: 
+=== "EIP1559"
+    ```
+    Gas Used =（Base Fee + MaxPriorityFeePerGas) * Tx Weight / {{ networks.moonbase.tx_weight_to_gas_ratio }}
+    ```
+=== "Legacy"
+    ```
+    Gas Used = GasPrice * Tx Weight / {{ networks.moonbase.tx_weight_to_gas_ratio }}
+    ```
+=== "EIP2930"
+    ```
+    Gas Used = GasPrice * Tx Weight / {{ networks.moonbase.tx_weight_to_gas_ratio }}
+    ```
```

Comment:
should there be a space in Gas Used and GasPrice or should there be no spaces, or do they both have to be the way they are now?